src/config.rs

```diff
@@ -131,30 +131,35 @@ impl Config {
         Ok(())
     }
 
-    pub async fn check_for_timezone(self: Config) -> Result<Config, Error> {
+    pub async fn check_for_timezone(self) -> Result<Config, Error> {
         if self.timezone.is_none() {
-            let desc = "Please select your timezone";
-            let mut options = TZ_VARIANTS
-                .to_vec()
-                .iter()
-                .map(|v| v.to_string())
-                .collect::<Vec<String>>();
-            options.sort();
-
-            let tz = input::select(desc, options, self.mock_select)?;
-            let config = Config {
-                timezone: Some(tz),
-                ..self
-            };
-
-            config.clone().save().await?;
-
+            let config = self.set_timezone().await?;
             Ok(config)
```

Comment:
```suggestion
            self.set_timezone().await
```
This should work because `set_timezone` returns a `Result<Config, Error>`. If you omit the semicolon it will return the value of this statement.